puffin/src/lib.rs

```diff
@@ -809,31 +1200,26 @@ fn test_short_file_name() {
 /// }
 /// ```
 ///
-/// Overhead: around 210 ns on 2020 Intel MacBook Pro.
+/// Overhead: around 54 ns on Macbook Pro with Apple M1 Max.
 #[macro_export]
 macro_rules! profile_function {
     () => {
         $crate::profile_function!("");
     };
     ($data:expr) => {
         let _profiler_scope = if $crate::are_scopes_on() {
-            static mut _FUNCTION_NAME: &'static str = "";
-            static mut _LOCATION: &'static str = "";
-            static _INITITIALIZED: ::std::sync::Once = ::std::sync::Once::new();
-
-            #[allow(unsafe_code)]
-            // SAFETY: accessing the statics is safe because it is done in cojunction with `std::sync::Once``
-            let (function_name, location) = unsafe {
-                _INITITIALIZED.call_once(|| {
-                    _FUNCTION_NAME = $crate::current_function_name!().leak();
-                    _LOCATION = format!("{}:{}", $crate::current_file_name!(), line!()).leak();
+            static SCOPE_ID: std::sync::atomic::AtomicU32 = std::sync::atomic::AtomicU32::new(0);
```

Comment:
The `Once` should already be taking care of the synchronization right? So you could avoid the atomic here?

Reply:
The std::sync::Once itself doesn't handle synchronization of the data after the initialization. It ensures that the initialization code is executed only once right?

Reply:
I need to initialize this static but can not use unsafe to mutate it.

Reply:
oh maybe i should use OnceCell?

Reply:
`Once` docs say:
```
When this function returns, it is guaranteed that some initialization has run and completed (it might not be the closure specified). It is also guaranteed that any memory writes performed by the executed closure can be reliably observed by other threads at this point (there is a happens-before relation between the closure and code executing after the return).
```
So should be ok to do an unsafe write to a static?
As it was before.

Or, yes, `OnceCell`, or more likely `OnceLock` should work, and be cleaner.

Reply:
the problem is that unsafe is not allowed in macros as crates often disallow unsafe, and then every macro needs a comment.

Reply:
yes needs to get rid of the `unsafe`, we can't currently use latest puffin library in our own project (!) because of this. which is obviously a requirement for us :smile: 
- #168

Reply:
Ok. I guess `OnceLock` would still be ok then. It's unsafe would be hidden in the standard library

Reply:
Using oncelock now, oncecell ended up in a deadlock or -stuck- for benchmarks